From b98611bf5897cb7419cb9fbbf7b82cab1976b482 Mon Sep 17 00:00:00 2001
From: LaurensVoerman <l.voerman@rug.nl>
Date: Wed, 5 Nov 2025 11:48:03 +0100
Subject: [PATCH] dicomLibs for vcpkg

---
 src/osgPlugins/dicom/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/osgPlugins/dicom/CMakeLists.txt b/src/osgPlugins/dicom/CMakeLists.txt
index 0ed6b5592..e5e53c13a 100644
--- a/src/osgPlugins/dicom/CMakeLists.txt
+++ b/src/osgPlugins/dicom/CMakeLists.txt
@@ -5,7 +5,7 @@ IF  (DCMTK_FOUND)
 
     SET(TARGET_SRC ReaderWriterDICOM.cpp )
 
-    LINK_LIBRARIES(${DCMTK_LIBRARIES})
+    LINK_LIBRARIES(DCMTK::DCMTK)
 
     add_compile_options("${DCMTK_CMAKE_CXX_FLAGS}")
     set(CMAKE_CXX_STANDARD 17)
-- 
2.51.0.windows.2

