From 7500f23a88c92bdd843a4a64a8a7d9bc86a86cd7 Mon Sep 17 00:00:00 2001
From: LaurensVoerman <l.voerman@rug.nl>
Date: Fri, 21 Nov 2025 17:31:09 +0100
Subject: [PATCH] BoostLibBeforeTiff

---
 src/CMakeLists.txt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index e35de06..51944dd 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -226,7 +226,11 @@ target_link_libraries(${LIBLAS_LIB_NAME}
   ${GEOTIFF_LIBRARY}
   ${GDAL_LIBRARY}
   ${LASZIP_LIBRARY}
-  ${Boost_LIBRARIES})
+  $<$<NOT:$<CONFIG:DEBUG>>:${Boost_IOSTREAMS_LIBRARY_RELEASE}>;$<$<CONFIG:DEBUG>:${Boost_IOSTREAMS_LIBRARY_DEBUG}>
+  $<$<NOT:$<CONFIG:DEBUG>>:${Boost_PROGRAM_OPTIONS_LIBRARY_RELEASE}>;$<$<CONFIG:DEBUG>:${Boost_PROGRAM_OPTIONS_LIBRARY_DEBUG}>
+  $<$<NOT:$<CONFIG:DEBUG>>:${Boost_SERIALIZATION_LIBRARY_RELEASE}>;$<$<CONFIG:DEBUG>:${Boost_SERIALIZATION_LIBRARY_DEBUG}>
+  $<$<NOT:$<CONFIG:DEBUG>>:${Boost_THREAD_LIBRARY_RELEASE}>;$<$<CONFIG:DEBUG>:${Boost_THREAD_LIBRARY_DEBUG}>
+  )
 
 target_link_libraries(${LIBLAS_C_LIB_NAME}
   ${LIBLAS_LIB_NAME} )
-- 
2.51.0.windows.2

